tests/lib/rules/test-case-property-ordering.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * @fileoverview Requires the properties of a test case to be placed in a consistent order.
+ * @author 薛定谔的猫<hh_2013@foxmail.com>
+ */
+
+'use strict';
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const rule = require('../../../lib/rules/test-case-property-ordering');
+const RuleTester = require('eslint').RuleTester;
+
+// ------------------------------------------------------------------------------
+// Tests
+// ------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester();
+ruleTester.run('test-case-property-ordering', rule, {
+  valid: [
+    `
+      new RuleTester().run('foo', bar, {
+        valid: [
+          {
+            code: "foo",
+            output: "bar",
+            options: ["baz"],
+          },
+        ]
+      });
+    `,
+    `
+      new RuleTester().run('foo', bar, {
+        valid: [
+          {
+            code: "foo",
+            output: "bar",
+            options: ["baz"],
+            env: { es6: true },
+          },
+        ]
+      });
+    `,
+    {
+      code: `
+        new RuleTester().run('foo', bar, {
+          valid: [
+            {
+              code: "foo",
+              options: ["baz"],
+              output: "bar",
+            },
+          ]
+        });
+      `,
+      options: [['code', 'errors', 'options', 'output', 'parserOptions']],
+    },
+  ],
+
+  invalid: [
+    {
+      code: `
+        new RuleTester().run('foo', bar, {
+          valid: [
+            {
+              code: "foo",
+              options: ["baz"],
+              output: "bar",
+            },
+          ]
+        });
+      `,
+      errors: [{ message: 'The properties of a test case should be placed in a consistent order: [code, output, options].' }],
+    },
+    {
+      code: `
+        new RuleTester().run('foo', bar, {
+          valid: [
+            {
+              code: "foo",
+              env: { es6: true },
+              output: "bar",
+              options: ["baz"],
+            },
+          ]
+        });
+      `,
+      errors: [{ message: 'The properties of a test case should be placed in a consistent order: [code, output, options].' }],
```

Comment:
I think this error message is a bit misleading because `[code, output, options]` are in the correct order, but the problem is that `env` should be at the end. Maybe it would be better if the error message said `[code, output, options, env]`.